### PR TITLE
make separate CodeTab window appear to right or left of NetLogo App

### LINKED ISSUE #1907: Separate Code Window always appears in the center of the user's screen
Furthermore it is below the application window and can be completely hidden

### PR DESCRIPTION
**Pull request by brandesNW**
fixes #1907

### CODE REVIEW

**Review by LaCuneta — CHANGES_REQUESTED** [review]
Thanks for this!  The code mostly looks good to me, I'm glad it was a pretty small fix.  I tested on Linux and noticed a few oddities:

1. (With single monitor) If I maximize the NetLogo window and then pop out the code window, it (usually, see item 3) still goes behind the main NetLogo window and I have to move windows around to get it on top.
2. (With dual monitors - left/right) If I maximize the NetLogo window on the left monitor and pop out the code window, it goes to the right monitor.  That's not the biggest problem, it's just a little weird.  If I maximize the NetLogo window on the right monitor and pop out the code window, I get the same behavior as the single monitor above.
3. In many cases (but not all, and I can't figure why), when I pop out the code tab window it does not have focus for me to start typing code, even if it is in view.  I would expect to be able to just start typing after popping it out, no matter where the window shows up.

I think item 3 is really the most serious one to me.  Even if the windows are overlapping after I click the pop out button, I at least want to be able to see the code window and start typing code or moving it around.

Let me know if you have any questions.

File: netlogo-gui/src/main/app/CodeTabContainer.scala [hunk 1]
```diff
@@ -2,21 +2,45 @@
...
-class CodeTabContainer(owner:          Frame,
-                       codeTabbedPane: JTabbedPane) extends JFrame with LinkChild {
+// parent              CodeTabContainer's LinkParent (a Frame).
+// codeTabbedPane      CodeTabContainer's JTabbedPane
+// codeTabDimension    CodeTabContainer's Dimension (gives size)
+// horizontalSpacing:  Horizontal spacing between CodeTabContainer and parent
+class CodeTabContainer(parent:            Frame,
+                       codeTabbedPane:    JTabbedPane,
+                       codeTabDimension:  Dimension = new Dimension(600, 400),
+                       horizontalSpacing: Int = 10) extends JFrame with LinkChild {
```
[the remark this answers is not in the document]
Comment (brandesNW) [reply netlogo-gui/src/main/app/CodeTabContainer.scala:35]: I wanted to expose them for future use. For example we might allow them to be created by a config file. 
I could just set them as local variables if you prefer.

[the remark this answers is not in the document]
Comment (LaCuneta) [reply netlogo-gui/src/main/app/CodeTabContainer.scala:35]: Leaving them in is fine, I just wanted to make sure the change was intentional.

File: netlogo-gui/src/main/app/CodeTabsPanel.scala [hunk 2]
```diff
@@ -131,4 +131,11 @@ class CodeTabsPanel(workspace:            GUIWorkspace,
       new AppEvents.SwitchedTabsEvent(previousTab, currentTab).raise(this)
     }
   }
+  java.awt.EventQueue.invokeLater(new Runnable() {
+
```
Comment (LaCuneta) [inline netlogo-gui/src/main/app/CodeTabsPanel.scala:135]: I'm not sure the blank space after the `new Runnable() {` is helpful here, I'd prefer it be between the end of the method above and the `java.awt.EventQueue` line.

File: netlogo-gui/src/main/app/CodeTabsPanel.scala [hunk 3]
```diff
@@ -131,4 +131,11 @@ class CodeTabsPanel(workspace:            GUIWorkspace,
...
+    override def run(): Unit = {
+        codeTabContainer.toFront()
+        codeTabContainer.repaint()
+    }
+})
```
Comment (LaCuneta) [inline netlogo-gui/src/main/app/CodeTabsPanel.scala:140]: The spacing here looks to be at 4 spaces, not 2.  The closing `})` is mis-aligned and should be indented 2 spaces to match up with the `(new Runnable() {` opening level.

**Review by LaCuneta — CHANGES_REQUESTED** [review]
The behavior was much better, I was able to type code every time and way I popped out the code tab window, and it never popped behind the main NetLogo window.

I noticed a few whitespace issues I left comments on.  Once those are fixed this should be good to merge.

**Review by LaCuneta — APPROVED** [review]

**Pull request merged**